### PR TITLE
Commenting widgets

### PR DESCRIPTION
**Pull request by abought**
## Ticket
https://openscience.atlassian.net/browse/EOSF-20

Replaces and extends PR #41 

# Purpose
- Turn comment code on file template into a component (`comments-pane`) made out of smaller components (`comment-detail` and `comment-form`)

screen shot 2016-07-01 at 2 20 18 pm

# Notes for Reviewers
## Testing notes
- Widget should support a variety of common comment actions:
  - [x]  create
  - [x] edit
  - [x] delete
  - [x] restore
  - ~~report/unreport?~~ (separate ticket)
- It should add newest comments to the top of the list- the same sort order used by the API.
- If you are the person who made a comment, you should see edit and delete buttons
  - If you are the owner of a deleted comment, you should see a restore button (but no text)
- If you are not the owner of a comment, you should see a report button
  - If you are not the owner of a deleted comment, you should not see comment text
- Comments on different things should be kept separate, eg project comments pane should not display the comments on a file owned by that project. (even though they use the same API endpoint)

## Known issues
- Initial pageload currently limited to 10 items (pagination).
- Cannot report comments (separate PR).=
- No support for threaded conversations. Emphasis is on model functionality.
- Doesn't check whether user can comment on project.

## Routes Added/Updated
- The project detail page has a new Comments section: http://localhost:4200/nodes/f263w
- The file detail page has a new Comments section: http://localhost:4200/nodes/3t9r8/files/osfstorage/576989df0dc31001ecbd4ef3

### CODE REVIEW

**Comment by abought** [comment]
Bringing this PR up to date. 

Some relationship error messages; investigating now. 

Got the following message when adding a comment to a file:
```
vendor.js:46009 TypeError: Cannot read property 'undefinedcomments' of undefined
    at ManyRelationship.removeCanonicalRecord (vendor.js:150725)
    at ManyRelationship.removeCanonicalRecords (vendor.js:150714)
    at ManyRelationship.computeChanges (vendor.js:150523)
    at ManyRelationship.updateRecordsFromAdapter (vendor.js:150847)
    at vendor.js:150546
```

And to a project:
```
TypeError: Cannot read property 'get' of undefined
    at ManyRelationship.removeCanonicalRecordFromInverse (vendor.js:150778)
    at ManyRelationship.removeCanonicalRecord (vendor.js:150723)
    at ManyRelationship.removeCanonicalRecords (vendor.js:150714)
    at ManyRelationship.computeChanges (vendor.js:150523)
    at ManyRelationship.updateRecordsFromAdapter (vendor.js:150847)
    at vendor.js:150546
```

File: addon/models/registration.js [hunk 1]
```diff
@@ -30,7 +30,11 @@ export default Node.extend({
         inverse: null
     }),
     contributors: DS.hasMany('contributors'),
-    comments: DS.hasMany('comments'),
+    comments: DS.hasMany('comments', {
+        updateRequest: {
+            requestType: () => 'POST'
```
Comment (samchrisinger) [inline addon/models/registration.js]: With newest `osf-adapter`, custom options to the relationship field (`updateRequest`) are no longer used.

**Comment by samchrisinger** [comment]
One comment, otherwise lgtm

File: addon/models/registration.js (hunk 1 above)
Comment (abought) [inline addon/models/registration.js]: That's good news! Changes pushed and CI passing.

**Pull request merged**